py/desimodel/inputs/throughput.py

```diff
@@ -0,0 +1,149 @@
+    params['area']['geometric_area'] *= 100**2
+
+    #- Load atmospheric extinction
+    d = fits.getdata(findfile('inputs/throughput/ZenithExtinction-KPNO.fits'), 'EXTINCTION')
+    extinction = InterpolatedUnivariateSpline(d['WAVELENGTH'], d['EXTINCTION'])
+
+    #- Load pre-spectrograph throughputs
+    thru = load_throughput(master_thru_file)
+
+    #- Load pre-computed fiberloss for reference objects
+    fiberinput = dict()
+    for objtype in ['elg', 'lrg', 'sky', 'star']:
+        fiberinput[objtype] = load_fiberinput(
+            findfile('throughput/fiberloss-{}.dat'.format(objtype)) )
+
+    #- Spectrograph throughputs
+    specthru = dict()
+
+    #- Min/Max wavelength coverage
+    wmin = dict()
+    wmax = dict()
+
+    for channel in ('b', 'r', 'z'):
+        specthru[channel] = load_spec_throughput(ccd_thru_file[channel])
+        wmin[channel], wmax[channel] = get_waveminmax(findfile('specpsf/psf-{}.fits'.format(channel)))
+
+        dw = 0.1
+        ww = np.arange(wmin[channel], wmax[channel]+dw/2, dw)
+        tt = thru(ww) * specthru[channel](ww)
+
+        data = np.rec.fromarrays([ww, tt, extinction(ww), fiberinput['elg'](ww)],
+                                names='wavelength,throughput,extinction,fiberinput')
+
+        hdr = fits.Header()
+        hdr['EXPTIME']  = (params['exptime_dark'], 'default exposure time [sec]')
+        hdr['GEOMAREA'] = (params['area']['geometric_area'], 'geometric area of mirror - obscurations')
+        hdr['FIBERDIA'] = (params['fibers']['diameter_arcsec'], 'average fiber diameter [arcsec]')
+        hdr['WAVEMIN']  = (wmin[channel], 'Minimum wavelength [Angstroms]')
+        hdr['WAVEMAX']  = (wmax[channel], 'Maximum wavelength [Angstroms]')
+
+        fiberinput_data = np.rec.fromarrays([ww, fiberinput['elg'](ww), fiberinput['lrg'](ww),
+                                      fiberinput['star'](ww), fiberinput['sky'](ww)],
+                                names='wavelength,elg,lrg,star,sky')
+
+        if outdir is None:
+            outdir = os.path.join(datadir(), 'throughput')
+
+        outfile = outdir + '/thru-{0}.fits'.format(channel)
+
+        hdus = fits.HDUList()
+        hdus.append(fits.PrimaryHDU())
+        hdus.append(fits.BinTableHDU(data, hdr, name='THROUGHPUT'))
+        hdus.append(fits.BinTableHDU(fiberinput_data, name='FIBERINPUT'))
+        hdus.writeto(outfile, clobber=True)
+
+
+def load_throughput(filename):
+    """
+    Load throughputs from DESI-0347, removing the spectrograph contributions
+    which will be loaded separately from higher resolution data.
+
+    Args:
+        filename: DESI-0347 Excel file location
+
+    Returns InterpolatedUnivariateSpline instance of thru vs. wave[Angstroms]
+    """
+    wave = docdb.xls_read_row(filename, 'Throughput', 3, 'C', 'P')*10
+    thru = docdb.xls_read_row(filename, 'Throughput', 112, 'C', 'P')
+    specthru = docdb.xls_read_row(filename, 'Throughput', 93, 'C', 'P')
+
```

Comment:
Since the excel file is hand maintained, we could get burned by hard coding the cell ranges like this in case they change or typos are introduced.  Adding some sanity checks on the extracted arrays would offer some protection, e.g., `assert np.all(np.diff(wave) > 0)`.